detekt-test-assertj/build.gradle.kts

```
@@ -0,0 +1,9 @@
+plugins {
+    id("module")
+    id("public-api")
+}
+
+dependencies {
+    compileOnly(projects.detektApi)
+    compileOnly(libs.assertj.core)
```

```suggestion
    implementation(libs.assertj.core)
```
It's reasonable to add this transitive dependency. Though, similar to how we treat the AGP dependency in DGP, we may want to lower the dependency version to the earliest that's compatible, and allow users to add their own runtime dependency on a newer version if they choose.

We'll have to do the same with the JUnit module since JUnit 6 only runs on JVM 17 and higher, and there's no reason to force consumers of that module to be on more modern versions.

---

#8533